examples_tests/20.Megatexture/main.cpp

```diff
@@ -704,46 +403,37 @@ int main()
         //phys addr texture doesnt have mips anyway and page table is accessed with texelFetch only
         params.MipmapMode = asset::ISampler::ESMM_NEAREST;
         params.TextureWrapU = params.TextureWrapV = params.TextureWrapW = asset::ISampler::ETC_CLAMP_TO_EDGE;
-        auto sampler = core::make_smart_refctd_ptr<asset::ICPUSampler>(params); //driver->createGPUSampler(params);
+        auto sampler = core::make_smart_refctd_ptr<asset::ICPUSampler>(params);
 
-        std::array<core::smart_refctd_ptr<asset::ICPUSampler>,ETP_COUNT> samplers;
+        std::array<core::smart_refctd_ptr<asset::ICPUSampler>,VT_COUNT> samplers;
         std::fill(samplers.begin(), samplers.end(), sampler);
 
         params.AnisotropicFilter = 0u;
         params.MaxFilter = asset::ISampler::ETF_NEAREST;
         params.MinFilter = asset::ISampler::ETF_NEAREST;
         params.MipmapMode = asset::ISampler::ESMM_NEAREST;
-        auto samplerPgt = core::make_smart_refctd_ptr<asset::ICPUSampler>(params); //driver->createGPUSampler(params);
-
-        std::array<core::smart_refctd_ptr<asset::ICPUSampler>, ETP_COUNT> samplersPgt;
-        std::fill(samplersPgt.begin(), samplersPgt.end(), samplerPgt);
-
-        std::array<asset::ICPUDescriptorSetLayout::SBinding, 2> bindings;
-        //page tables
-        bindings[0].binding = 0u;
-        bindings[0].count = ETP_COUNT;
-        bindings[0].stageFlags = asset::ISpecializedShader::ESS_FRAGMENT;
-        bindings[0].type = asset::EDT_COMBINED_IMAGE_SAMPLER;
-        bindings[0].samplers = samplersPgt.data();
-
-        //physical addr textures
-        bindings[1].binding = 1u;
-        bindings[1].count = ETP_COUNT;
-        bindings[1].stageFlags = asset::ISpecializedShader::ESS_FRAGMENT;
-        bindings[1].type = asset::EDT_COMBINED_IMAGE_SAMPLER;
-        bindings[1].samplers = samplers.data();
-
-        ds0layout = core::make_smart_refctd_ptr<asset::ICPUDescriptorSetLayout>(bindings.data(), bindings.data()+bindings.size());
+        auto samplerPgt = core::make_smart_refctd_ptr<asset::ICPUSampler>(params);
+
+        auto bindings = vt->getDSlayoutBindings(PGTAB_BINDING, PHYSICAL_STORAGE_VIEWS_BINDING);
+        (*bindings)[0].samplers = &samplerPgt;
+        (*bindings)[1].samplers = samplers.data();
+
+        ds0layout = core::make_smart_refctd_ptr<asset::ICPUDescriptorSetLayout>(bindings->data(), bindings->data()+bindings->size());
     }
     core::smart_refctd_ptr<asset::ICPUDescriptorSetLayout> ds2layout;
```

Comment:
this isn't really relevant to the overall VT API, but why not put everything to do with VT together in one descriptor set?

Reply:
just because it's easier this way

Reply:
but if I wanted to do that I would be able to achieve that?

Reply:
decent idea, lets not create spurious `dynamic_array`s